.github/workflows/main.yaml

```diff
@@ -21,3 +27,126 @@ jobs:
         run: poetry install --no-interaction --all-extras
       - name: run pre-commit
         run: poetry run pre-commit run --all-files --show-diff-on-failure
+  packaging:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+      - name: Install dependencies
+        run: poetry install --no-interaction --all-extras
+      - name: build package
+        run: poetry build
+  documentation:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+      - name: Install dependencies
+        run: poetry install --no-interaction --all-extras
+      - name: build documentation
+        run: poetry run make html --directory docs
+  unit-tests:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        python: ["3.8", "3.9", "3.10", "3.11", "3.12"]
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+      - name: Install dependencies
+        run: poetry install --no-interaction --all-extras
+      - name: Setup Python
+        id: setup-python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python }}
+          cache: 'poetry'
+      - name: run python tests
+        run: poetry run tox -e py
+      - name: run python test report
+        run: poetry run tox -e report
+  docker-build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+      - name: Build Docker Image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+  integration-tests:
+    needs:
+      - unit-tests
+    if: ${{ github.event_name != 'pull_request' }}
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+      - name: Install dependencies
+        run: poetry install --no-interaction --all-extras
+      # Update poetry for https://github.com/python-poetry/poetry/issues/7184
+      - name: update poetry
+        run: poetry self update --no-ansi
+      - name: Setup Python
+        id: setup-python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
+          cache: 'poetry'
+      - name: Assume AWS role
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: ${{ env.AWS_REGION }}
+          role-to-assume: ${{ env.AWS_ROLE }}
+          role-session-name: GHA-${{ github.repository_owner }}-${{ github.event.repository.name }}-${{ github.run_id }}
+          role-duration-seconds: ${{ env.AWS_ROLE_DURATION }}
+      - name: run tests
+        run: poetry run behave integration-tests/features --junit --junit-directory build/behave
```

Comment:
ugh! i think something has changed in AWS which is causing some of these tests to fail now.  It's now producing errors like `WaitConditionHandle does not support any properties`.  I'm guessing that it's related to [aws-cloudformation issue #1097](https://github.com/aws-cloudformation/cloudformation-coverage-roadmap/issues/1097).

Circleci builds have been turned off because two systems cannot run integration tests against the same AWS account.  I think we should comment out integration tests for now so that we at least have GH action testing in place for new PRs.   The integration test fix might be more involved so I suggest we fix those in a follow on PR.